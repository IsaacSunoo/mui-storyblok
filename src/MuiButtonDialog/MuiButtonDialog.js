import React, {
  createElement,
  useState,
  lazy,
  Suspense,
} from 'react';
import PropTypes from 'prop-types';

const MuiButton = lazy(() => import('../MuiButton/MuiButton'));
const MuiDialog = lazy(() => import('../MuiDialog/MuiDialog'));

/**
 * MuiButtonDialog controls Dialog open state from true to false.
 * More docs and demos at https://material-ui.com/api/button/
 * Uses MuiButton and MuiDialog
 */

export const MuiButtonDialog = ({
  dialog,
  button,
}) => {
  const components = {
    MuiDialog,
    MuiButton,
  };

  const [state, setState] = useState({ open: false });
  const toggleDialog = () => setState({ ...state, open: !state.open });
  const muidialog = dialog[0];
  const muibutton = button[0];

  return (
    <Suspense fallback={<div />}>
      {
      muibutton
        ? createElement(
          components[muibutton.component],
          { ...muibutton, onClick: toggleDialog },
        )
        : null
    }
      {
      muidialog
        ? createElement(
          components[muidialog.component],
          { ...muidialog, open: state.open, toggleDialog },
        )
        : null
    }
    </Suspense>
  );
};

export default MuiButtonDialog;

MuiButtonDialog.propTypes = {
  /** MuiButton Allowed maximum: 1 */
  button: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,

  /** MuiDialog Allowed maximum: 1 */
  dialog: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,

};

MuiButtonDialog.defaultProps = {};
