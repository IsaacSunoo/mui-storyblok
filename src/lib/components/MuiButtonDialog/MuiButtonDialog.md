MuiButton example:

```js noeditor
import MuiButtonDialog from './MuiButtonDialog';
```

```js
import MuiButtonDialog from './MuiButtonDialog';

const dialog = [{
    component: 'MuiDialog',
    content: [{
      component: 'MuiDialogContent',
      content: [{
        component: 'MuiTypography',
        content: [{
          component: 'MuiText',
          text: 'text',
        }],
      }],
    }],
    dialogTitle: [{
      component: 'MuiDialogTitle',
      text: 'text',
    }],
  }];

const button = [{
    component: 'MuiButton',
    buttonText: 'buttonText',
}];

  <MuiButtonDialog
    dialog={dialog}
    button={button}
  />
```