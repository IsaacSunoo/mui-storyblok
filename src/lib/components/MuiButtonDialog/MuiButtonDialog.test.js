import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiButtonDialog from './MuiButtonDialog';

function setup() {
  const props = {
    button: [{
      component: 'MuiButton',
      buttonText: 'buttonText',
    }],
    dialog: [{
      component: 'MuiDialog',
      content: [{
        component: 'MuiDialogContent',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            text: 'text',
          }],
        }],
      }, {
        component: 'MuiDialogActions',
        content: [{
          component: 'MuiButtonRedirect',
          href: 'google.com',
          fileName: 'fileName.png',
          button: [{
            component: 'MuiButton',
            buttonText: 'here',
          }],
        }],
      }],
      dialogTitle: [{
        component: 'MuiDialogTitle',
        text: 'text',
        toggleDialog: jest.fn(),
      }],
    }],
  };
  const comp = mount(<MuiButtonDialog {...props} />);

  return { comp, props };
}

describe('<MuiButtonDialog />', () => {
  it('renders MuiButtonDialog', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiButtonDialog {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
