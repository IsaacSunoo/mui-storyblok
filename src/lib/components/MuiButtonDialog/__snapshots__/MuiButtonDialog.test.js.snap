// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiButtonDialog /> snapshot 1`] = `<div />`;
